vertx-consul-client/src/main/java/io/vertx/ext/consul/ConsulClient.java

```diff
@@ -0,0 +1,607 @@
+   */
+  @Fluent
+  ConsulClient createSession(Handler<AsyncResult<String>> idHandler);
+
+  /**
+   * Initialize a new session
+   *
+   * @param options   options used to create session
+   * @param idHandler will be provided with ID of new session
+   * @return reference to this, for fluency
+   * @see <a href="https://www.consul.io/docs/agent/http/session.html#session_create">/v1/session/create</a> endpoint
+   */
+  @Fluent
+  ConsulClient createSessionWithOptions(SessionOptions options, Handler<AsyncResult<String>> idHandler);
+
+  /**
+   * Returns the requested session information
+   *
+   * @param id            the ID of requested session
+   * @param resultHandler will be provided with info of requested session
+   * @return reference to this, for fluency
+   * @see <a href="https://www.consul.io/docs/agent/http/session.html#session_info">/v1/session/info/&lt;session&gt;</a> endpoint
+   */
+  @Fluent
+  ConsulClient infoSession(String id, Handler<AsyncResult<Session>> resultHandler);
+
+  /**
+   * Renews the given session. This is used with sessions that have a TTL, and it extends the expiration by the TTL
+   *
+   * @param id            the ID of session that should be renewed
+   * @param resultHandler will be provided with info of renewed session
+   * @return reference to this, for fluency
+   * @see <a href="https://www.consul.io/docs/agent/http/session.html#session_renew">/v1/session/renew/&lt;session&gt;</a> endpoint
+   */
+  @Fluent
+  ConsulClient renewSession(String id, Handler<AsyncResult<Session>> resultHandler);
+
+  /**
+   * Returns the active sessions
+   *
+   * @param resultHandler will be provided with list of sessions
+   * @return reference to this, for fluency
+   * @see <a href="https://www.consul.io/docs/agent/http/session.html#session_list">/v1/session/list</a> endpoint
+   */
+  @Fluent
+  ConsulClient listSessions(Handler<AsyncResult<List<Session>>> resultHandler);
+
+  /**
+   * Returns the active sessions for a given node
+   *
+   * @param nodeId        the ID of node
+   * @param resultHandler will be provided with list of sessions
+   * @return reference to this, for fluency
+   * @see <a href="https://www.consul.io/docs/agent/http/session.html#session_node">/v1/session/node/&lt;node&gt;</a> endpoint
+   */
+  @Fluent
+  ConsulClient listNodeSessions(String nodeId, Handler<AsyncResult<List<Session>>> resultHandler);
+
+  /**
+   * Destroys the given session
+   *
+   * @param id            the ID of session
+   * @param resultHandler will be called when complete
+   * @return reference to this, for fluency
+   * @see <a href="https://www.consul.io/docs/agent/http/session.html#session_destroy">/v1/session/destroy/&lt;session&gt;</a> endpoint
+   */
+  @Fluent
+  ConsulClient destroySession(String id, Handler<AsyncResult<Void>> resultHandler);
+
+  /**
+   * Close the client and release its resources
+   */
+  void close();
```

Comment:
This method just closes underlying `HttpClient`. Delegated method is synchronous and I believe this one should be also sync